test-common/test-common-api/src/test/java/org/apache/kafka/common/test/api/ClusterTestExtensionsTest.java

```diff
@@ -331,4 +342,105 @@ public void testControllerListenerName(ClusterInstance cluster) throws Execution
             assertEquals(1, admin.describeMetadataQuorum().quorumInfo().get().nodes().size());
         }
     }
+
+    @ClusterTest(
+        types = {Type.KRAFT, Type.CO_KRAFT},
+        brokerSecurityProtocol = SecurityProtocol.SASL_PLAINTEXT,
+        controllerSecurityProtocol = SecurityProtocol.SASL_PLAINTEXT,
+        serverProperties = {
+            @ClusterConfigProperty(key = GroupCoordinatorConfig.OFFSETS_TOPIC_PARTITIONS_CONFIG, value = "1"),
+            @ClusterConfigProperty(key = GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, value = "1")
+        }
+    )
+    public void testSaslPlaintext(ClusterInstance clusterInstance) throws InterruptedException {
+        Assertions.assertEquals(SecurityProtocol.SASL_PLAINTEXT, clusterInstance.config().brokerSecurityProtocol());
+
+        // default ClusterInstance#admin helper with admin credentials
+        try (Admin admin = clusterInstance.admin()) {
+            assertDoesNotThrow(() -> admin.describeAcls(AclBindingFilter.ANY).values().get());
+        }
+        String topic = "sasl-plaintext-topic";
+        Assertions.assertDoesNotThrow(() -> clusterInstance.createTopic(topic, 1, (short) 1));
+        try (Producer<byte[], byte[]> producer = clusterInstance.producer()) {
+            assertDoesNotThrow(() -> producer.send(new ProducerRecord<>(topic, Utils.utf8("key"), Utils.utf8("value"))).get());
+            assertDoesNotThrow(producer::flush);
+        }
+        try (Consumer<byte[], byte[]> consumer = clusterInstance.consumer()) {
+            consumer.subscribe(List.of(topic));
+            TestUtils.waitForCondition(() -> {
+                ConsumerRecords<byte[], byte[]> records = consumer.poll(Duration.ofMillis(100));
+                return records.count() == 1;
+            }, "Failed to receive message");
+        }
+
+        // client with non-admin credentials
+        Map<String, Object> nonAdminConfig = Map.of(
```

Comment:
the helper methods of `clusterInstance` should be able configure the `sasl.jaas.config` automatically - that is why we add those helper to `clusterInstance`

Reply:
Hi @chia7712, thanks for the review. Yes, we already set default user to `admin`, `consumer`, and `producer` helper function automatically. Here, we would like to test developer overrides `sasl.jaas.config` with non-admin credentials. We also test overriding with unknown credential, so we can make sure the cluster instance with sasl plaintext works.